src/components/inputs/input.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { forwardRef } from 'react';
+import { View, TextInput, Text, TextInputProps, TextStyle } from 'react-native';
+
+import { useInputStyles } from './input.styles';
+
+interface InputProps extends Omit<TextInputProps, 'style'> {
+  disabled?: boolean;
+  endEnhancer?: React.ReactElement | string;
+  error?: string;
+  handleInputRef?: (ref: TextInput) => void;
+  index?: number;
+  startEnhancer?: React.ReactElement | string;
+  testId?: string;
+  textAlign?: Exclude<TextStyle['textAlign'], 'auto' | 'justify'>;
+  isPassword?: boolean;
```

Comment:
This seems odd. If I am building an email component, I do not expect this property.

I believe you want to create PasswordInput using this. But Input should not know anything about password.